code/datums/emotes.dm

```diff
@@ -0,0 +1,152 @@
+#define EMOTE_VISIBLE 1
+#define EMOTE_AUDIBLE 2
+
+/datum/emote
+	var/key = "" //What calls the emote
+	var/key_third_person = "" //This will also call the emote
+	var/message = "" //Message displayed when emote is used
+	var/message_param = "" //Message to display if a param was given
+	var/emote_type = EMOTE_VISIBLE //Whether the emote is visible or audible
+	var/restraint_check = FALSE //Checks if the mob is restrained before performing the emote
+	var/muzzle_ignore = FALSE //Will only work if the emote is EMOTE_AUDIBLE
+	var/muzzled_noise = "" // For snowflaking the muzzled noise
+	var/list/mob_type_blacklist_typecache //Types that are NOT allowed to use that emote
+	var/punct = "."
```

Comment:
What is the point of this? Why isn't it just embedded in the message var of emotes that want punctuation?

Reply:
For automatically creating directed emotes (paradise has a lot of these). For example, for "snap", you only need to define:
`message = "snaps"`
And if you snap at someone, it'll automatically do: 
`User snaps at someone.`